app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseDatabase.kt

```diff
@@ -297,6 +300,69 @@ class FirebaseDatabase(activity: Activity) : Database {
     override suspend fun unfollow(follower: String, followee: String) {
         doFollow(follower, followee, follow = false)
     }
-}
 
+    /**
+     * Returns a list of users that liked a given challenge
+     * @param cid the challenge id
+     * @return a list of users that liked the challenge
+     */
+    internal fun getLikeRefById(cid: String): LazyRef<User> {
+        return object : BaseLazyRef<User>() {
+            override fun fetchValue(): Task<User> {
+                return dbChallengeRef.child(cid).child("likes").get().thenMap {
+                    val uid = it.key!!
+                    getUserById(uid).fetch().result
```

Comment:
HMMMM. Are you sure that this work. This is a DataSnapshot and the method key returns either "likes" or "cid" which does not correspond to any valid user id. Also please update the comment (it does not return a list of users)

I believe what you want to do is first convert the DataSnapshot to a `Map<String, Boolean>` and then collect each key such that the corresponding value is sets to `true`. These are the id of all of the users that like the challenge

Reply:
Check `doFollow` method from Dario.

Reply:
This function is implemented differently now as I have decided to change the way I'd like to implement this functionality. The new function works because I have written specifically tests to ensure that it is doing its task properly.